ni/bpo.py

```diff
@@ -35,9 +36,14 @@ def __init__(self, server: ni_abc.ServerHost) -> None:
         elif any(x not in (True, False, None) for x in status_results):
             raise TypeError("unexpected value in " + str(status_results))
 
-        if all(status_results):
-            return ni_abc.Status.signed
-        elif any(value is None for value in status_results):
-            return ni_abc.Status.username_not_found
-        else:
-            return ni_abc.Status.not_signed
+        failures = {
+            None: ni_abc.Status.username_not_found,
+            False: ni_abc.Status.not_signed,
+        }
+        problems: Mapping[ni_abc.Status, Set[str]] = defaultdict(set)
+        for username, result in results.items():
+            if result:
+                continue
+            problems[failures[result]].add(username)
```

Comment:
Another way to do this which alleviates the need for `defaultdict` is:
```suggestion
            problems.setdefault(failures[result], set()).add(username)
```
This way if anyone chooses to rely on `KeyError` to detect any potential failures it will continue to work as expected versus having to check if the returned set if empty.

Reply:
Ah, I must admit I didn't remember about `setdefault`, thanks for the feedback.